media-libs/libplacebo/libplacebo-2.72.0.ebuild

```diff
@@ -0,0 +1,68 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_REQ_USE="xml"
+PYTHON_COMPAT=( python3_{6,7,8,9} )
+
+if [[ "${PV}" == "9999" ]]; then
+	EGIT_REPO_URI="https://code.videolan.org/videolan/libplacebo.git"
+	inherit git-r3
+else
+	KEYWORDS="~amd64 ~ppc64 ~x86"
+	SRC_URI="https://code.videolan.org/videolan/libplacebo/-/archive/v${PV}/libplacebo-v${PV}.tar.gz"
+	S="${WORKDIR}/${PN}-v${PV}"
+fi
+
+inherit meson multilib-minimal python-any-r1
+
+DESCRIPTION="Reusable library for GPU-accelerated image processing primitives"
+HOMEPAGE="https://code.videolan.org/videolan/libplacebo"
+
+LICENSE="LGPL-2.1+"
+SLOT="0/$(ver_cut 2)" # libplacebo.so version
+IUSE="glslang lcms +opengl +shaderc test +vulkan"
+REQUIRED_USE="vulkan? ( || ( glslang shaderc ) )"
+
+RDEPEND="glslang? ( dev-util/glslang[${MULTILIB_USEDEP}] )
+	lcms? ( media-libs/lcms:2[${MULTILIB_USEDEP}] )
+	opengl? ( media-libs/libepoxy[${MULTILIB_USEDEP}] )
+	shaderc? ( >=media-libs/shaderc-2017.2[${MULTILIB_USEDEP}] )
+	vulkan? ( media-libs/vulkan-loader[${MULTILIB_USEDEP}] )"
+DEPEND="${RDEPEND}"
+
+BDEPEND="virtual/pkgconfig
+	vulkan? (
+		${PYTHON_DEPS}
+		dev-util/vulkan-headers
+		$(python_gen_any_dep "dev-python/mako[\${PYTHON_USEDEP}]")
```

Comment:
This is missing `python_check_deps` as documented at https://dev.gentoo.org/~mgorny/python-guide/any.html#dependencies

Reply:
Also it's easier to read if you single-quote the dependency here, so you don't have to escape the `$`

Reply:
Hmm.. reading this doc, one thing I still find myself wondering is whether `python-any-r1` is even the correct ebuild to be using? The way libplacebo's build system is coded, it re-uses the same python interpreter that `meson` is being run by (which is the default behaviour of meson's `python` module when you don't specify a specific version of python).

So rather than generating a check for *any* python version, we need the `dev-python/mako` check on the exact version of python that is being used by `dev-util/meson`. Problem is: I don't know how to code that check, since seeing as `dev-util/meson` doesn't use `python-single`, **it depends on what interpreter the user has configured as their system default interpreter**.

It's possible that the proper resolution to this would be to allow users to somehow override the version of python being used when compiling libplacebo, but that probably requires a .1 release implementing this functionality, or a custom `.patch`, or otherwise.

Perhaps the sanest resolution is to simply ignore the issue and just generate a blanket `dev-python/mako` dependency, without specifying any particular version, and hoping it works out.

Thoughts?

Reply:
If meson is calling the "default" python interpreter, then `python-any-r1.eclass` will handle it properly for you. It sets the "default" python interpreter to the first one for which `python_check_deps` passes. The `python` command defaults to the system default, but can be overridden by setting the `EPYTHON` environment variable to a specific one. This is what the eclass will do in `python-utils-r1_pkg_setup`, which gets called automatically if nothing else defines a `pkg_setup`.

Reply:
Aah, that makes sense. Thanks, updated the PR.